sopel/plugins/rules.py

```diff
@@ -732,58 +741,58 @@ def is_unblockable(self) -> bool:
         """
 
     @abc.abstractmethod
-    def is_user_rate_limited(self, nick: Identifier) -> bool:
+    def is_user_rate_limited(self, nick: Identifier, at_time: Optional[datetime.datetime] = None) -> bool:
         """Tell when the rule reached the ``nick``'s rate limit.
 
-        :return: ``True`` when the rule reached the limit, ``False`` otherwise
+        :param nick: the nick associated with this check.
+        :param at_time: optional aware datetime for the rate limit check. If not given, ``utcnow`` will be used.
+        :return: ``True`` when the rule reached the limit, ``False`` otherwise.
         """
 
     @abc.abstractmethod
-    def is_channel_rate_limited(self, channel: Identifier) -> bool:
+    def is_channel_rate_limited(self, channel: Identifier, at_time: Optional[datetime.datetime] = None) -> bool:
         """Tell when the rule reached the ``channel``'s rate limit.
 
-        :return: ``True`` when the rule reached the limit, ``False`` otherwise
+        :param channel: the channel associated with this check.
+        :param at_time: optional aware datetime for the rate limit check. If not given, ``utcnow`` will be used.
+        :return: ``True`` when the rule reached the limit, ``False`` otherwise.
         """
 
     @abc.abstractmethod
-    def is_global_rate_limited(self) -> bool:
+    def is_global_rate_limited(self, at_time: Optional[datetime.datetime] = None) -> bool:
         """Tell when the rule reached the global rate limit.
 
-        :return: ``True`` when the rule reached the limit, ``False`` otherwise
+        :param at_time: optional aware datetime for the rate limit check. If not given, ``utcnow`` will be used.
```

Comment:
```suggestion
        :param at_time: optional aware datetime for the rate limit check.
                        If not given, ``utcnow`` will be used.
```

Reply:
(see my reply to the previous `at_time` param's line note)

Reply:
(same)